Improve the OAuth index page

hiddewie opened the pull request:
Some minor improvement for the OAuth token/application page:
- Open authorized application links in a new tab
- Make some English titles use `OAuth` instead of `oauth`.
- Add a line about the expiration of tokens.
- Use monospace for displaying OAuth tokens

~~It might be a good idea to hide (or even delete tokens) that are no longer valid.~~

Screenshot:

![image](https://user-images.githubusercontent.com/1073881/96349796-338c7c00-10b2-11eb-9c3c-1bae0baf7525.png)

tomhughes commented:
What tokens do you think we are showing that are no longer valid?

hiddewie commented:
I assumed that all tokens had an expiration date. So any granted access would automatically be revoked after a certain time. This seems not to be the case.

Many applications ask (For example the HOT tasking manager) ask every for a new access token every X days (not sure how many). That is why I thought the tokens expired automatically, and I wanted to see how long each token would be valid.

This would be good to mention in text on the page.

mmd-osm commented:
I guess what you're seeing here with the HOT tasking manager has been discussed here before: https://github.com/openstreetmap/openstreetmap-website/issues/1455

Also, note that we're still on OAuth 1.0a at this time.

hiddewie commented:
Hmm indeed, thanks for finding that issue.

OK I removed the `token.valid_to` visualization because the tokens do not expire. I added a line of text under the table that tells how long the tokens are valid.

gravitystorm commented on app/views/oauth_clients/index.html.erb:
```diff
@@ -15,7 +15,7 @@
   <% @tokens.each do |token| %>
     <tr>
-      <td><%= link_to token.client_application.name, token.client_application.url %></td>
+      <td><%= link_to token.client_application.name, token.client_application.url, target: "_blank" %></td>
```
We rarely use `target: "_blank"` for links - any particular reason to add it to this one?

Also this line causes the CI failures for both hash rocket syntax, and noopener when using target: "_blank"

hiddewie replied:
I would argue that any link that does not remain in the current application should be opened in a new tab. This makes users less likely to lose their way if they click on the OAuth application name. There are many more external links which should maybe also be opened in a new tab. It would be even better to annotate them with an icon like below to notify users of such links.

![image](https://user-images.githubusercontent.com/1073881/97108386-8cb96880-16cd-11eb-9e3d-5b3ea7c940c3.png)

hiddewie commented:
@gravitystorm I added the noopener and fixed the linting.

simonpoole commented:
A tiny bit of nitpicking with the original text on the page that should be fixed in a follow up PR. The text currently says

`The following tokens have been issued to applications in your name:`

and then goes on to not display tokens but an application list. 

`Tokens have been issued in your name for the following applications;`

which would be more correct (as always this will require changing the id of the string, not a simply text change, to invalidate translations).

mmd-osm commented:
The list is indeed based on individual tokens and shows the respective application rather than any token details:

https://github.com/openstreetmap/openstreetmap-website/blob/master/app/views/oauth_clients/index.html.erb#L16

mmd-osm commented:
As we have OAuth 2.0 in the meantime and OAuth 1.0a will be deprecated and removed eventually, I‘d vote to close this one.

If there’s anything worth improving on the OAuth 2 pages, focus on that instead.

hiddewie commented:
>  I‘d vote to close this one.

Fine for me :+1:

danieldegroot2 commented:
Switch to OAuth 2, from OAuth 1 and HTTP Basic Auth, is being tracked at https://github.com/openstreetmap/operations/issues/867
